### PR TITLE
Hide preview in Recent Apps & ReAuthenticate on UserInactivity

### LINKED ISSUE #107: Hide Ivy Wallet preview in Recent Apps when "App Lock" is ON
You got Biometrics but my account is still exposed. People can see from the background my money.

Also:
- When "App Lock" is ON automatically navigate to authentication screen again after 5min of inactivity

### PR DESCRIPTION
**Pull request by Vishwa-Raghavendra**
## Pull Request (PR) Checklist

Please check if your pull request fulfills the following requirements:

- [x] The PR is submitted to the `develop` branch.
- [x] I've read the **[Contribution Guidelines](https://github.com/ILIYANGERMANOV/ivy-wallet/blob/main/CONTRIBUTING.md)**.
- [x] The code builds and is tested on an actual Android device.
- [x] I confirm that I've run the code locally and everything works as expected.

_Put an `x` in the boxes that apply._

- [x] Demo: Checking checkbox using `[x]`

## Pull Request Type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Small update (fix typo, change color or something)
- [ ] Gradle Build related changes
- [ ] Dependencies update (updating libraries)
- [ ] Documentation
- [x] Other (please describe): Code Improvement
_Put an `x` in the boxes that apply._

## Does this PR closes any GitHub Issues?

Check **[Ivy Wallet Issues](https://github.com/ILIYANGERMANOV/ivy-wallet/issues)**.

- Closes #107

## What's changed?

Describe with a few bullets **what's new:**

- With this commit Users with "App Lock Enabled" gets their Ivy wallet preview hidden in recent menu (Users can still take screenshot within the app)

- After 5 Mins of inactivity the users are prompted to re-authenticate

### Code Changes
This commit includes code improvements which are
- New Screen is added to Screen class
- authenticateWithOSBiometricsModal() functional changes
- handleBiometricAuthenticationResult() changes
- UserInactivity is now managed via Ivy Context

### Testing Devices

- Redmi Note 10 (Android 11)
- Moto g40 (Android 11)
- Redmi Note 7 pro (Android 10)

Works as expected on the above devices

_This PR template style was inspired by [ionic-framework](https://github.com/ionic-team/ionic-framework/blob/main/.github/PULL_REQUEST_TEMPLATE.md)._

### CODE REVIEW

**Comment by ILIYANGERMANOV** [comment]
Hi @Vishwa-Raghavendra, thank you for your pull request! At first glance all seems good 👍  I'll review and test it tomorrow morning :)

File: app/src/main/java/com/ivy/wallet/ui/IvyContext.kt [hunk 1]
```diff
@@ -215,4 +220,40 @@ class IvyContext {
...
+        if (userInactiveJob != null && userInactiveJob!!.isActive)
+            return
+
+        userInactiveJob = GlobalScope.launch(Dispatchers.IO) {
+            while (userInactiveTime.get() < USER_INACTIVE_TIME_LIMIT  && userInactiveJob != null && !userInactiveJob?.isCancelled!!) {
+                delay(1000)
+                userInactiveTime.incrementAndGet()
+            }
+            if (!isUserInactive.value)
+                _isUserInactive.value = (true)
```
Comment (ILIYANGERMANOV) [inline app/src/main/java/com/ivy/wallet/ui/IvyContext.kt:246]: `(true)` isn't needed. It can be just `_isUserInactive.value = true`

**Review by ILIYANGERMANOV — APPROVED** [review]

**Comment by ILIYANGERMANOV** [comment]
Hi @Vishwa-Raghavendra this is a very good PR! I like it :)

A few minor comments:
- Ivy Wallet's code style ({}, etc) is a bit different but that's my bad because I didn't provide you a document.
- Overall it's a good solution and I liked your code improvements

I'll merge your PR now. Also please provide me a **link to your LinkedIn profile** so I can endorse you for Android Development and give you **Ivy Wallet Premium Lifetime access code** there :) 

Thank you for your contribution! It's much appreciated.
P.S. The Contributors Wall will be auto-updated 24h after you commit enters the `main` branch.

**Comment by Vishwa-Raghavendra** [comment]
Hi ILIYANGERMANOV,

Thank you for the quick responses. Here is the link to my LinkedIn profile

Link --> https://www.linkedin.com/in/vishwa-raghavendra

I apologize for the delay in responding as I was down on health for the
last couple of days.
Thank you for accepting the PR :)

Regards
Vishwa

**Comment by ILIYANGERMANOV** [comment]
Hi @Vishwa-Raghavendra I hope that you're well now! I sent you a connection request with the Ivy Premium access code on LinkedIn.

For some very weird reason Github didn't add you to the contributors graph :/ I guess it's because I had to do force-push in `develop`. Can you submit a new empty PR which I'll approve so you'll be displayed as a contributor like you deserve :)

**Comment by Vishwa-Raghavendra** [comment]
Hi ILIYANGERMANOV,

I wasn't able to create an empty PR hence I have updated the recycler view
library from 1.2.0 -> 1.2.1 and raised a pull request.

**Pull request merged**